packages/backend/src/workers/action.ts

```diff
@@ -30,66 +30,87 @@ type JobData = {
-      async (err) => {
-        // this happens when the prerequisite steps for the action fails (e.g. db error, missing execution, flow, step, etc...)
-        // in such cases, we do not want to retry
-        await Execution.setStatus(jobData.executionId, 'failure')
-        throw new UnrecoverableError(err.message || 'Action failed to execute')
-      },
-    )
-
-    if (executionStep.isFailed) {
-      await Execution.setStatus(executionId, 'failure')
-      return handleErrorAndThrow(executionStep.errorDetails, executionError)
-    }
+      span?.addTags({
+        flowId: jobData.flowId,
+        executionId: jobData.executionId,
+        stepId: jobData.stepId,
+        actionKey: step?.key,
+        appKey: step?.appKey,
+      })
 
-    if (!nextStep) {
-      await Execution.setStatus(executionId, 'success')
-      return
-    }
+      const {
+        flowId,
+        executionId,
+        nextStep,
+        executionStep,
+        nextStepMetadata,
+        executionError,
+      } = await processAction({ ...jobData, jobId: job.id }).catch(
+        async (err) => {
+          // this happens when the prerequisite steps for the action fails (e.g. db error, missing execution, flow, step, etc...)
+          // in such cases, we do not want to retry
+          await Execution.setStatus(jobData.executionId, 'failure')
+          throw new UnrecoverableError(
+            err.message || 'Action failed to execute',
+          )
+        },
+      )
+
+      if (executionStep.isFailed) {
+        await Execution.setStatus(executionId, 'failure')
+        return handleErrorAndThrow(executionStep.errorDetails, executionError)
+      }
 
-    const jobName = `${executionId}-${nextStep.id}`
+      if (!nextStep) {
+        await Execution.setStatus(executionId, 'success')
+        return
+      }
 
-    const jobPayload = {
-      flowId,
-      executionId,
-      stepId: nextStep.id,
-      metadata: nextStepMetadata,
-    }
+      const jobName = `${executionId}-${nextStep.id}`
+
+      const jobPayload = {
+        flowId,
+        executionId,
+        stepId: nextStep.id,
+        metadata: nextStepMetadata,
+      }
 
-    let jobOptions = DEFAULT_JOB_OPTIONS
+      let jobOptions = DEFAULT_JOB_OPTIONS
 
-    if (step.appKey === 'delay') {
-      jobOptions = {
-        ...DEFAULT_JOB_OPTIONS,
-        delay: delayAsMilliseconds(step.key, executionStep.dataOut),
+      if (step.appKey === 'delay') {
+        jobOptions = {
+          ...DEFAULT_JOB_OPTIONS,
+          delay: delayAsMilliseconds(step.key, executionStep.dataOut),
+        }
       }
-    }
 
-    await actionQueue.add(jobName, jobPayload, jobOptions)
+      await actionQueue.add(jobName, jobPayload, jobOptions)
+    } catch (e) {
+      const isRetriable =
+        !(e instanceof UnrecoverableError) &&
+        job.attemptsMade < MAXIMUM_JOB_ATTEMPTS
+
+      if (isRetriable) {
+        span?.addTags({
+          will_retry: 'true',
```

Comment:
nit: standardize to `camelCase` can

```suggestion
          willRetry: 'true',
```